From 732f5939ef2f57d002839fca28785416fbdd08e5 Mon Sep 17 00:00:00 2001
From: Yong Gan <yong.gan@nxp.com>
Date: Wed, 22 Aug 2018 07:16:02 +0800
Subject: [PATCH 4427/5242] MGS-4130 [#imx-1133] Cannot pass the stress test
 of low bus suspend/resume.

commit  e7f9a7b27b39639ca01df8efefd1f4a85eea242e from
https://source.codeaurora.org/external/imx/linux-imx.git

Add delay time and retry to make sure the Power state was changed.

Signed-off-by Yong Gan <yong.gan@ncp.com>

(cherry picked from commit 59a836f35807f5c09c98cf3be318ac49526bcf69)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../hal/os/linux/kernel/gc_hal_kernel_driver.c     |   16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c
index 4e0c419..39beea7 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c
@@ -1169,7 +1169,21 @@ static int gpu_resume(struct platform_device *dev)
             else
 #endif
             {
-                status = gckHARDWARE_SetPowerManagementState(device->kernels[i]->hardware, statesStored);
+                gctINT j = 0;
+
+                for (; j < 100; j++)
+                {
+                    status = gckHARDWARE_SetPowerManagementState(device->kernels[i]->hardware, statesStored);
+
+                    if (( statesStored != gcvPOWER_OFF_BROADCAST
+                       && statesStored != gcvPOWER_SUSPEND_BROADCAST)
+                       || status != gcvSTATUS_CHIP_NOT_READY)
+                    {
+                        break;
+                    }
+
+                    gcmkVERIFY_OK(gckOS_Delay(device->kernels[i]->os, 10));
+                };
             }
 
             if (gcmIS_ERROR(status))
-- 
1.7.9.5

